### ISSUE TITLE
Test cases for inventory layout

### ISSUE THREAD

**Issue by herlo** [CentOS-PaaS-SIG/linchpin#583]
Write some tests around verifying inventory layout. Consider template data, complex structures in host_group:vars, etc.

**Comment by herlo** [comment]
Inventory Layout Example 1

```
inventory_layout:
#  inventory_file: cnv-inventory.inv
  vars:
    hostname: __IP__
  hosts:
    "{{ executor_name }}-master":
      count: 1
      host_groups:
        - masters
        - etcd
        - nodes
        - nfs
        - glusterfs
    "{{ executor_name }}-node":
      count: 2
      host_groups:
        - nodes
        - glusterfs
  host_groups:
    OSEv3:
      children:
        - masters
        - nodes
        - etcd
        - nfs
        - glusterfs
      vars:
        ansible_ssh_user: root
        containerized: true
        openshift_version: "3.9.25"
        openshift_release: "v3.9"
        openshift_image_tag: "v3.9.25"
        openshift_schedulable: true
        osm_default_node_selector: ""
        openshift_node_labels: "{'region': 'infra','zone': 'default', 'node-role.kubernetes.io/compute': 'true'}"
        openshift_deployment_type: openshift-enterprise
        openshift_clock_enabled: true
        openshift_master_identity_providers: "[{'name': 'allow_all_auth', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]"
        openshift_disable_check: "memory_availability,disk_availability"
        openshift_enable_excluders: false
        template_service_broker_install: false
        openshift_use_manageiq: false
        openshift_install_examples: false

        openshift_hosted_etcd_storage_kind: nfs
        openshift_hosted_etcd_storage_nfs_options: "*(rw,root_squash,sync,no_wdelay)"
        openshift_hosted_etcd_storage_nfs_directory: "/opt/osev3-etcd"
        openshift_hosted_etcd_storage_volume_name: "etcd-vol2"
        openshift_hosted_etcd_storage_access_modes: "['ReadWriteOnce']"
        openshift_hosted_etcd_storage_volume_size: "1100M"
        openshift_hosted_etcd_storage_labels: "{'storage': 'etcd'}"

        ansible_service_broker_local_registry_whitelist: "[\".*-apb$\"]"

        docker_dev: "/dev/vdb"
        glusterfs_devices: "['/dev/vdc']"

        # webconsole
        openshift_web_console_prefix: "jniederm/origin-"
        openshift_web_console_version: "demo-v10"

        # cns
        openshift_storage_glusterfs_namespace: "app-storage"
        openshift_storage_glusterfs_storageclass: true
        openshift_storageclass_default: false
        openshift_storage_glusterfs_storageclass_default: true
        openshift_storage_glusterfs_block_deploy: false
```

**Comment by samvarankashyap** [comment]
i have couple of them which you would like to try out : 
```yaml
inventory_layout:
#  inventory_file: cnv-inventory.inv
  vars:
    hostname: __IP__
  hosts:
    "master":
      count: 1
      host_groups:
        - masters
        - etcd
        - nodes
        - nfs
    "node":
      count: 2
      host_groups:
        - nodes
  host_groups:
    OSEv3:
      children:
        - masters
        - nodes
        - etcd
        - nfs
      vars:
        ansible_ssh_user: root
        openshift_schedulable: true
        openshift_node_labels: "{'region': 'infra','zone': 'default', 'node-role.kubernetes.io/compute': 'true'}"
        openshift_deployment_type: openshift-enterprise
        openshift_clock_enabled: true
        openshift_master_identity_providers: "[{'name': 'allow_all_auth', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]"
        openshift_disable_check: "memory_availability,disk_availability"
        openshift_enable_excluders: false
        template_service_broker_install: false
        openshift_use_manageiq: false
        openshift_install_examples: false
 
        openshift_hosted_etcd_storage_kind: nfs
        openshift_hosted_etcd_storage_nfs_options: "*(rw,root_squash,sync,no_wdelay)"
        openshift_hosted_etcd_storage_nfs_directory: "/opt/osev3-etcd"
        openshift_hosted_etcd_storage_volume_name: "etcd-vol2"
        openshift_hosted_etcd_storage_access_modes: "['ReadWriteOnce']"
        openshift_hosted_etcd_storage_volume_size: "1100M"
        openshift_hosted_etcd_storage_labels: "{'storage': 'etcd'}"
 
        ansible_service_broker_refresh_interval: "20s"
        ansible_service_broker_local_registry_whitelist: "[\".*-apb$\"]"
        ansible_service_broker_image_tag: "latest"
 
        ansible_service_broker_etcd_image_prefix: "quay.io/coreos/"
        ansible_service_broker_etcd_image_tag: "latest"
 
        docker_dev: "/dev/vdb"
 
        osm_default_node_selector: ""
```

**Comment by samvarankashyap** [comment]
```yaml
inventory_layout:
#  inventory_file: cnv-inventory.inv
  vars:
    hostname: __IP__
  hosts:
    "somename-master":
      count: 1
      host_groups:
        - masters
        - etcd
        - nodes
        - nfs
        - glusterfs
    "somename-node":
      count: 2
      host_groups:
        - nodes
        - glusterfs
  host_groups:
    OSEv3:
      children:
        - masters
        - nodes
        - etcd
        - nfs
        - glusterfs
      vars:
        ansible_ssh_user: root
        containerized: true
        openshift_version: "3.9.25"
        openshift_release: "v3.9"
        openshift_image_tag: "v3.9.25"
        openshift_schedulable: true
        osm_default_node_selector: ""
        openshift_node_labels: "{'region': 'infra','zone': 'default', 'node-role.kubernetes.io/compute': 'true'}"
        openshift_deployment_type: openshift-enterprise
        openshift_clock_enabled: true
        openshift_master_identity_providers: "[{'name': 'allow_all_auth', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]"
        openshift_disable_check: "memory_availability,disk_availability"
        openshift_enable_excluders: false
        template_service_broker_install: false
        openshift_use_manageiq: false
        openshift_install_examples: false
        openshift_hosted_etcd_storage_kind: nfs
        openshift_hosted_etcd_storage_nfs_options: "*(rw,root_squash,sync,no_wdelay)"
        openshift_hosted_etcd_storage_nfs_directory: "/opt/osev3-etcd"
        openshift_hosted_etcd_storage_volume_name: "etcd-vol2"
        openshift_hosted_etcd_storage_access_modes: "['ReadWriteOnce']"
        openshift_hosted_etcd_storage_volume_size: "1100M"
        openshift_hosted_etcd_storage_labels: "{'storage': 'etcd'}"
        ansible_service_broker_local_registry_whitelist: "[\".*-apb$\"]"
        docker_dev: "/dev/vdb"
        glusterfs_devices: "['/dev/vdc']"
        # webconsole
        openshift_web_console_prefix: "jniederm/origin-"
        openshift_web_console_version: "demo-v10"
        # cns
        openshift_storage_glusterfs_namespace: "app-storage"
        openshift_storage_glusterfs_storageclass: true
        openshift_storageclass_default: false
        openshift_storage_glusterfs_storageclass_default: true
        openshift_storage_glusterfs_block_deploy: false
```

**Comment by samvarankashyap** [comment]
@herlo there are very subtle changes in both of the above inventory layouts however the regression occurred when using them ^^^ should be resolved by  PR #581

**Comment by herlo** [comment]
Fixed by #587
